File: chap1_arrays_and_hashing/src/ht_main.py
# Rename to main.py

from table import HashTable

if __name__ == "__main__":
    ht = HashTable(8)
    ht.set("apple", 3)
    ht.set("banana", 7)
    ht.set("orange", 5)
    ht.set("banana", 10)  # update

    try:
        print("banana =>", ht.get("banana"))
    except KeyError:
        print("banana not found")

    ht.remove("apple")
    try:
        print("apple =>", ht.get("apple"))
    except KeyError:
        print("apple not found")

File: chap1_arrays_and_hashing/src/ht_main.c
// Rename this file to main.c

#include "table.h"
#include <stdio.h>

int main(void) {
    HashTable *ht = ht_create(10);
    ht_set(ht, "apple", 3);
    ht_set(ht, "banana", 7);
    ht_set(ht, "orange", 5);
    ht_set(ht, "banana", 10); // update

    int found;
    int val = ht_get(ht, "banana", &found);
    printf("banana => %d (found? %d)\n", val, found);

    ht_remove(ht, "apple");
    val = ht_get(ht, "apple", &found);
    printf("apple => %d (found? %d)\n", val, found);

    ht_free(ht);
    return 0;
}
